File: 2000/5/7-5.py
#!/usr/bin/env python

import sys

class Point:
    def __init__(self):
        self.x = 0.0
        self.y = 0.0

def distance(a, b):
    return ((a.x - b.x) ** 2 + (a.y - b.y) ** 2) ** 0.5

cnt = 1
while True:
    line = sys.stdin.readline()
    if line == "":
        break
    ships = line.split()
    ships = int(ships[0])
    plot = []
    for i in xrange(ships):
        line = sys.stdin.readline()
        line = map(float, line.split())
        pt = Point()
        pt.x = line[0]
        pt.y = line[1]
        plot.append(pt)
    mini = distance(plot[0], plot[1])
    n1 = 0
    n2 = 1
    for i in xrange(ships - 1):
        for j in xrange(i + 1, ships):
            tmp = distance(plot[i], plot[j])
            if tmp < mini:
                mini = tmp
                n1 = i
                n2 = j
    print "Fleet %s: Ship %s and Ship %s are the closeest pair." % (cnt, n1 + 1, n2 + 1)
    cnt += 1

File: 2000/5/7-5.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath>
using namespace std;

struct Point
{
	double x, y;
	Point() : x(0), y(0)
	{}
};

double distance(Point a, Point b)
{
	return sqrt(pow(a.x - b.x, 2) + pow(a.y - b.y, 2));
}

void main()
{
	ifstream in("prob5_in.txt");
	int ships, n1, n2, cnt = 1;
	double min, tmp;
	vector<Point> plot;
	while (in >> ships)
	{
		plot.reserve(ships);
		for (int i = 0; i < ships; i++)
			in >> plot[i].x >> plot[i].y;
		min = distance(plot[0], plot[1]);
		n1 = 0;
		n2 = 1;
		for (i = 0; i < ships - 1; i++)
		{
			for (int j = i + 1; j < ships; j++)
			{
				tmp = distance(plot[i], plot[j]);
				if (tmp < min)
				{
					min = tmp;
					n1 = i;
					n2 = j;
				}
			}
		}
		cout << "Fleet " << cnt++ << ": Ship " << n1 + 1 << " and Ship " << n2 + 1
			<< " are the closest pair." << endl;
		plot.clear();
	}
}
